mobile/mobile.go

```diff
@@ -18,41 +18,54 @@ import (
 
 var log = logging.MustGetLogger("mobile")
 
+// Messenger is used to inform the bridge layer of new data waiting to be queried
+type Messenger interface {
+	Notify(update *tcore.ThreadUpdate)
+}
+
 // Wrapper is the object exposed in the frameworks
 type Wrapper struct {
-	RepoPath string
+	RepoPath  string
+	messenger Messenger
+}
+
+// NodeConfig is used to configure the mobile node
+type NodeConfig struct {
+	RepoPath      string
+	CentralApiURL string
+	LogLevel      string
 }
 
 // NewNode is the mobile entry point for creating a node
 // NOTE: logLevel is one of: CRITICAL ERROR WARNING NOTICE INFO DEBUG
-func NewNode(repoPath string, centralApiURL string, logLevel string) (*Wrapper, error) {
+func NewNode(config *NodeConfig, messenger Messenger) (*Wrapper, error) {
 	var m Mobile
-	return m.NewNode(repoPath, centralApiURL, logLevel)
+	return m.NewNode(config, messenger)
 }
 
 // Mobile is the name of the framework (must match package name)
 type Mobile struct{}
 
 // Create a gomobile compatible wrapper around TextileNode
-func (m *Mobile) NewNode(repoPath string, centralApiURL string, logLevel string) (*Wrapper, error) {
```

Comment:
So how is log level working now?

Reply:
It's the same but you set `LogLevel` on the config object instead. There were getting to be many constructor params.